### PR TITLE
[Network] Virtual-wan support multi link

### PR DESCRIPTION
**Pull request by 00Kai0**
closed https://github.com/Azure/azure-cli/issues/19233
closed https://github.com/Azure/azure-cli-extensions/issues/4074

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/virtual-wan/azext_vwan/_params.py [hunk 1]
```diff
@@ -161,12 +185,25 @@ def load_arguments(self, _):
...
         c.argument('address_prefixes', nargs='+', help='Space-separated list of CIDR address prefixes.')
+        c.argument('with_link', help='Create VPN site with default link.', arg_type=get_three_state_flag())
 
     with self.argument_context('network vpn-site', arg_group='Device Property') as c:
         c.argument('device_model', help='Model of the device.')
         c.argument('device_vendor', help='Name of the device vendor.')
         c.argument('link_speed', help='Link speed in Mbps.', type=int)
 
+    with self.argument_context('network vpn-site link') as c:
+        c.argument('vpn_site_link_name', help='The name of vpn site link.')
```
Comment (kairu-ms) [inline src/virtual-wan/azext_vwan/_params.py:196]: The optional of this argument should be '--name' and '-n'

File: src/virtual-wan/azext_vwan/_params.py [hunk 2]
```diff
@@ -130,12 +130,36 @@ def load_arguments(self, _):
...
+        c.argument("vpn_site_link", help='The resource ID of VPN Site Link.')
+        c.argument('with_link', help='Create VpnConnection with default link.', arg_type=get_three_state_flag())
 
     with self.argument_context('network vpn-gateway connection list') as c:
         # List commands cannot use --ids flag
         c.argument('resource_name', vpn_gateway_name_type, id_part=None)
         c.argument('gateway_name', id_part=None)
 
+    with self.argument_context('network vpn-gateway connection vpn-site-link-conn') as c:
+        c.argument("vpn_site_link_conn_name", help='Name of the VPN gateway connection.', id_part='child_name_1')
```
Comment (kairu-ms) [inline src/virtual-wan/azext_vwan/_params.py:142]: The optional of this argument should be '--name' and '-n'

File: linter_exclusions.yml [hunk 3]
```diff
@@ -1627,6 +1627,38 @@ network vhub connection create:
...
+      - option_length_too_long
+    vpn_connection_protocol_type:
+      rule_exclusions:
+      - option_length_too_long
+    vpn_link_connection_mode:
+      rule_exclusions:
+      - option_length_too_long
+    vpn_site_link_conn_name:
+      rule_exclusions:
+      - option_length_too_long
```
Comment (kairu-ms) [inline linter_exclusions.yml:1646]: please use '--name' and '-n' for this argument under `network vpn-gateway connection vpn-site-link-conn` command group

**Review by kairu-ms — COMMENTED** [review]
@00Kai0 There's no more issues in code. Could you add examples in _help.py. Especially add examples for `--with-link` argument usage.

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**